dev/gallery.tsx

```diff
@@ -189,9 +217,10 @@ export function Gallery() {
                                 ? graphTypeContainsText(question, search)
                                 : true,
                         )
+                        .map(insertShowTooltips)
                         .map(([question, i]) => (
                             <QuestionRenderer
-                                key={i}
+                                key={`${i}${showTooltips ? "-with-tooltips" : ""}`}
```

Comment:
Adding text to the key to force a re-render when toggling the "Show Tooltips" switch.